File: sliptrack-backend/src/main/java/com/sliptrack/sliptrackbackend/model/PaymentSlip.java
package com.sliptrack.sliptrackbackend.model;

import com.sliptrack.sliptrackbackend.enums.PaymentStatus;
import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Entity
@Table(name = "payment_slips")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class PaymentSlip {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String iban;

    @Column(nullable = false)
    private BigDecimal amount;

    private String referenceNumber;

    private String paymentModel;

    private String payerName;

    private String description;

    private LocalDate dueDate;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    @Builder.Default
    private PaymentStatus status = PaymentStatus.UNPAID;

    private String imageUrl;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "category_id", nullable = false)
    private Category category;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "property_id")
    private Property property;

    private LocalDateTime scannedAt;

    @CreationTimestamp
    @Column(nullable = false, updatable = false)
    private LocalDateTime createdAt;
}

File: sliptrack-mobile/src/types/paymentSlip.ts
export type PaymentStatus = "PAID" | "UNPAID";

export interface PaymentSlip {
  id: number;
  iban: string;
  amount: number;
  referenceNumber: string | null;
  paymentModel: string | null;
  providerName: string | null;
  description: string | null;
  dueDate: string;
  status: PaymentStatus;
  paidAt: string | null;
  imageUrl: string | null;
  categoryId: number | null;
  categoryName: string | null;
  subCategoryId: number | null;
  subCategoryName: string | null;
  propertyId: number | null;
  propertyName: string | null;
  scannedAt: string | null;
  createdAt: string;
}

export interface PaymentSlipFilters {
  status?: PaymentStatus;
  categoryId?: number;
  subCategoryId?: number;
  propertyId?: number;
  dueDateFrom?: string;
  dueDateTo?: string;
}

export interface PaymentSlipRequest {
  iban: string;
  amount: number;
  referenceNumber?: string;
  paymentModel?: string;
  providerName?: string;
  description?: string;
  dueDate: string;
  categoryId: number;
  subCategoryId?: number;
  propertyId?: number;
  wasScanned?: boolean;
}
